syntax = "proto3";

package Monitor;

service MonitorService {
  // The monitor is in local network, and put the monitor stream to the server
  rpc PutMonitorStream (MonitorRequest) returns (stream MonitorData) {}
  rpc PutMonitorSummary (MonitorRequest) returns (MonitorSummary) {}
  rpc GetOperationStream (stream MonitorRequest) returns (stream MonitorOperation) {}
}

message MonitorRequest {
  bool operation = 1;
}

message MonitorData {
  string filename = 1;
  string data = 2;
}

message MonitorSummary {
  string start_time = 1;
  string end_time = 2;
  string data = 3;
}

message MonitorOperation {
  string mode = 1;
  int32 interval = 2;
}